resource_impl.h

```diff
@@ -23,26 +23,106 @@
 #include <errno.h>
 
 #define RESOURCE_64	64
+#define RESOURCE_256	256
 #define RESOURCE_2048	2048
 
+#define FNAMELEN	RESOURCE_256
+
+#define INITSCOPE	"/init.scope"
+
+/* We are assuming that cgroup is mounted at its usual location,
+ * If we have reeust from applications who do not mount cgroup at
+ * usual location, then we need to change this.
+ */
+#define DEFAULTCGFS	"/sys/fs/cgroup"
+
 #define eprintf(msg, ...)	fprintf(stderr,\
 					"Err at line %d in file %s: "msg"\n",\
 					__LINE__, __FILE__, ##__VA_ARGS__)\
 
+static inline void clean_init(char *cg)
+{
+	char *p;
+	size_t cg_len = strlen(cg), init_len = strlen(INITSCOPE);
+
+	if (cg_len < init_len)
+		return;
+
+	p = cg + cg_len - init_len;
+	if (strcmp(p, INITSCOPE) == 0) {
+		if (p == cg)
+			*(p+1) = '\0';
+		else
+			*p = '\0';
+	}
+}
+
+/* Get cgroup path for a particular controller */
+static inline char *get_cgroup(pid_t pid, const char *contrl)
```

Comment:
Have you looked at LXCs cgroup parsing?

Reply:
No, however this parsing method is inspired from LXCFS parsing which I had a look at.
Is there something specific which I should have taken care of in the method?

Reply:
It's just more stable, tested and better maintained I'd argue. :)

Reply:
I looked at it, my version was missing the controller list scan in case of multiple controller cgroup. I added that. I did not copy whole function as I did not want to do realloc etc which might slow the performance for application which use libresource for monitoring purposes. Let me know if that is ok.